front/lib/swr/actions.ts

```diff
@@ -25,3 +28,40 @@ export function useAssistantConfigurationActions(
     error,
   };
 }
+
+export function useAgentConfigurationActions(
+  ownerId: string,
+  agentConfigurationId: string | null,
+  mcpServerViews?: Array<{ sId: string; server: { name: string } }>
+) {
+  const disabled = agentConfigurationId === null;
+  const actionsFetcher: Fetcher<GetActionsResponseBody> = fetcher;
+  const { data, error } = useSWRWithDefaults(
+    `/api/w/${ownerId}/builder/assistants/${agentConfigurationId}/actions`,
+    actionsFetcher,
+    {
+      disabled,
+      revalidateOnFocus: false,
+      revalidateOnReconnect: false,
+    }
+  );
+
+  // Transform assistant builder actions to agent builder form actions
+  const actions = useMemo((): AgentBuilderAction[] => {
+    if (!data?.actions) {
+      return [];
+    }
+
+    const transformResult = transformAssistantBuilderActionsToFormData(
+      data.actions,
+      mcpServerViews || []
```

Comment:
_nit/curiosity_: I'm wondering if in that use using our helper `emptyArray` can make sure we don't return a different empty array, possibly make some extra rendering.